### ISSUE TITLE
Form fields with errors should have aria-describedby or similar markup

### ISSUE THREAD

**Issue by toolness** [18F/django-uswds-forms#6]
The USWDS documentation has [example fields with form errors](https://standards.usa.gov/components/form-controls/#text-input) have an `aria-describedby` attribute that points back to the `<span>` containing the content of the errors relevant to that field. This is particularly nice for screen readers because it allows users to easily hear what's wrong with a form field when they focus on it.

It would be really nice if e.g. `{% fieldset %}` did this kind of thing automatically.

**Comment by toolness** [comment]
Hmm, it might be hard to tie `aria-describedby` directly back to the errors produced by the [`ErrorList` class](https://github.com/django/django/blob/master/django/forms/utils.py#L77), as the `ErrorList` just gets a list of string errors to display, with no way to tie them back to the form field they may have came from (assuming they're not non-field errors).

That said, we could potentially _duplicate_ the error information in a hidden element and tie _that_ to `aria-describedby` when we render the input element...

**Comment by toolness** [comment]
Another alternative may be to specify this information only in the `{% fieldset %}` tag; it could bypass our custom `ErrorList` subclass and instead render the errors on its own, giving them an ID, which it then connects `aria-describedby` to.

**Comment by tbaxter-18f** [comment]
I'm reaching here, but what about ErrorDict? It has a reference to field we might leverage?
